tests/ert/ui_tests/cli/test_field_parameter.py

```diff
@@ -476,32 +477,33 @@ def test_field_param_update_using_heat_equation_zero_var_params_and_adaptive_loc
 
         with warnings.catch_warnings(record=True) as record:
             warnings.simplefilter("always")  # Ensure all warnings are always recorded
-            smoother_update(
-                new_prior,
-                new_posterior,
-                experiment.observation_keys,
-                config.ensemble_config.parameters,
-                ObservationSettings(),
-                ESSettings(localization=True),
-            )
+            with caplog.at_level(logging.INFO):
+                smoother_update(
+                    new_prior,
+                    new_posterior,
+                    experiment.observation_keys,
+                    config.ensemble_config.parameters,
+                    ObservationSettings(),
+                    ESSettings(localization=True),
+                )
 
-            warning_messages = [(w.category, str(w.message)) for w in record]
+                # Note that this used to fail since run time and user warnings were
+                # thrown because we tried updating parameters with zero variance.
+                assert not record
 
-            # Check that each required warning appears at least once
-            assert any(
-                issubclass(w[0], RuntimeWarning)
-                and "divide by zero encountered in divide" in w[1]
-                for w in warning_messages
-            )
-            assert any(
-                issubclass(w[0], UserWarning)
-                and "Cross-correlation matrix has entries not in [-1, 1]" in w[1]
-                for w in warning_messages
-            )
+                assert (
+                    "There are 50 parameters with 0 variance that will not be updated."
+                    in caplog.text
+                )
 
         param_config = config.ensemble_config.parameter_configs["COND"]
         prior_result = new_prior.load_parameters("COND")["values"]
         posterior_result = new_posterior.load_parameters("COND")["values"]
+
+        # Make sure parameters with zero variance are not updated.
+        assert (prior_result.values[:, :, :5, 0] == 1).all()
```

Comment:
not sure that I understand this test or maybe I do but future generations might not :smile: 
can we just do?:
```python
(prior_result.values[:, :, :5, 0] ==posterior_result.values[:, :, :5, 0]).all()
```

Reply:
Nice, that looks better. I've pushed a fix.